### ISSUE TITLE
Remove prestop configuration from deployment.yaml

### ISSUE THREAD

**Issue by scottfrederick** [spring-guides/top-spring-on-kubernetes#8]
The following should be removed from the example `deployment.yaml` file: 

```
          lifecycle:
            preStop:
              exec:
                command: ["sh", "-c", "sleep 10"]
```

The `preStop` is not necessary since Spring Boot apps handle `SIGTERM` notifications and gracefully shut down. 

In addition, Docker images for applications are often built with distroless base images that do not include a shell, so commands like the `sh -c sleep 10` recommended here will not work. The `./mvnw spring-boot:build-image` command suggested in this guide will use a base image that does not include a shell [starting with Spring Boot 3.4](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.4.0-M1-Release-Notes#paketo-tiny-builder-for-building-oci-images).

**Comment by robertmcnees** [comment]
Thanks, @scottfrederick. Great catch. I have a pending PR #7 on this guide that touches a lot of files. I made an additional commit there that removes the `preStop` instructions.

PR #7 is waiting on a review from @ryanjbaxter but please feel free to also have a look.

**Comment by scottfrederick** [comment]
The credit goes to @anthonydahanne for finding this and bringing it to our attention. Thanks @robertmcnees for the cleanup PR, I'll take a look.

**Comment by robertmcnees** [comment]
This can likely be closed now that PR #7 is merged. Specifically [this commit](https://github.com/spring-guides/top-spring-on-kubernetes/pull/7/commits/f6fb9dfdde91d01ef77a60b58d31c72a2731d314) that was merged as part of that PR should solve the concern.
